br/pkg/lightning/mydump/region.go

```diff
@@ -442,23 +478,47 @@ func SplitLargeFile(
 				pos = dataFile.FileMeta.FileSize
 			}
 			endOffset = pos
-			parser.Close()
+			subParser.Close()
+		}
+		isChunkNeeded := true
+		var rowsToSkip int64
+		// iterate the overall parser to get how many lines to skip in this chunk
+		if remainingSkipRows > 0 {
+			iterPos, _ := parser.Pos()
+			for remainingSkipRows > 0 && iterPos < endOffset {
+				if err := parser.ReadRow(); err != nil {
```

Comment:
In https://github.com/pingcap/tidb/pull/40852 I use `parser.ReadUntilTerminator` because the skipped row may contain bad syntax